allensdk/brain_observatory/ecephys/align_timestamps/barcode_sync_dataset.py

```diff
@@ -0,0 +1,70 @@
+import warnings
+
+import numpy as np
+
+from . import barcode
+from allensdk.brain_observatory.ecephys.file_io.ecephys_sync_dataset import (
+    EcephysSyncDataset,
+)
+
+
+class BarcodeSyncDataset(EcephysSyncDataset):
+    @property
+    def barcode_line(self):
+        """ Obtain the index of the barcode line for this dataset.
+
+        """
+
+        if "barcode" in self.line_labels:
+            return self.line_labels.index("barcode")
+        elif "barcodes" in self.line_labels:
+            return self.line_labels.index("barcodes")
+        else:
+            raise ValueError("no barcode line found")
+
+    def extract_barcodes(self, **barcode_kwargs):
+        """ Read barcodes and their times from this dataset's barcode line.
+
+        Parameters
+        ----------
+        **barcode_kwargs : 
+            Will be passed to .barcode.extract_barcodes_from_times
+
+        Returns 
+        -------
+        times : np.ndarray
+            The start times of each detected barcode.
+        codes : np.ndarray
+            The values of each detected barcode
+
+        """
+
+        sample_freq_digital = float(self.sample_frequency)
+        barcode_channel = self.barcode_line
+
+        on_events = self.get_rising_edges(barcode_channel)
+        off_events = self.get_falling_edges(barcode_channel)
+
+        on_times = on_events / sample_freq_digital
```

Comment:
Any concerns about type division here?

Reply:
This is py3 only and timestamps are definitely supposed to be floats, so I think this is OK.